src/abstractdataframe/reshape.jl

```diff
@@ -48,54 +52,72 @@ d1s_name = stack(d1, Not([:a, :b, :e]), variable_name=:somemeasure)
 """
 function stack(df::AbstractDataFrame, measure_vars::AbstractVector{<:Integer},
                id_vars::AbstractVector{<:Integer}; variable_name::Symbol=:variable,
-               value_name::Symbol=:value, view::Bool=false)
+               value_name::Symbol=:value, view::Bool=false,
+               vareltype::Type=CategoricalString)
     if view
-        return _stackview(df, measure_vars, id_vars,
-                          variable_name=variable_name, value_name=value_name)
+        return _stackview(df, measure_vars, id_vars, variable_name=variable_name,
+                          value_name=value_name, vareltype=vareltype)
     end
     N = length(measure_vars)
-    cnames = names(df)[id_vars]
-    insert!(cnames, 1, value_name)
-    insert!(cnames, 1, variable_name)
-    DataFrame(AbstractVector[repeat(_names(df)[measure_vars], inner=nrow(df)), # variable
+    cnames = _names(df)[id_vars]
+    pushfirst!(cnames, value_name)
+    pushfirst!(cnames, variable_name)
+    if vareltype <: CategoricalString
+        nms = String.(_names(df)[measure_vars])
+        catnms = categorical(nms)
+        levels!(catnms, nms)
+    elseif vareltype <: Symbol
```

Comment:
```suggestion
    elseif vareltype === Symbol
```
(Same below.)

Reply:
OK, but I have always thought `<:` is the recommended style in such cases :).

Reply:
When I see `<:` I always wonder whether the RHS is an abstract type. But that's really a minor point, YMMV.